project/src/main/java/com/google/sps/Models/Database.java

```diff
@@ -183,4 +183,36 @@ public static Document getDocumentByHash(String hash) {
 
     return docs;
   }
+
+  //Takes in a Document hash and a User email
+  //Adds the userID to the Document's list of Users
+  //Adds the Document's hash to the User's list of Documents
+  //Returns true if successful, false if the user doesn't exist
+  public static boolean shareDocument(String hash, String email) {
+    Query query = new Query("User").addFilter("email", Query.FilterOperator.EQUAL, email);
+    Entity userEntity = getDatastore().prepare(query).asSingleEntity();
+
+    if(userEntity == null) {
```

Comment:
Something to consider: what if the shared person hasn't registered yet? It would be good to surface the error and (even better) set up some simple registration automation flow (e.g. send an email invite).

Reply:
I have validation to make sure it's an email on the front end and if there is no user with that email, then the conditional on line 195 will be true and the function will return false, in turn triggering the user does not exist response in the SharingServlet